app/models/user.rb

```diff
@@ -12,26 +12,23 @@ class User < ApplicationRecord
   has_many :friendships
   has_many :inverse_friendships, class_name: 'Friendship', foreign_key: 'friend_id'
 
+  has_many :pending_friendships, -> { where status: false }, class_name: 'Friendship', foreign_key: 'user_id'
+  has_many :pending_friends, through: :pending_friendships, source: :friend
+
+  has_many :requested_friends, -> { where status: false }, class_name: 'Friendship', foreign_key: 'friend_id'
+  has_many :friend_requests, through: :requested_friends, source: :user
+
   def friends
     friends_array = friendships.map { |friendship| friendship.friend if friendship.status } +
                     inverse_friendships.map { |friendship| friendship.user if friendship.status }
     friends_array.compact
   end
```

Comment:
**REQUIRED**
* Let's define user's friends using database queries since queries are designed to be more efficient.
Since we have two rows for mutual friendship, we can define users friends using user id only. Let's change this to:
```ruby
has_many :confirmed_friendships, -> { where status: true }, class_name: "Friendship"
has_many :friends, through: :confirmed_friendships, source: :user
```

Reply:
Fixed it